hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -259,9 +259,9 @@ AbfsThrottlingIntercept getIntercept() {
     return intercept;
   }
 
-  List<AbfsHttpHeader> createDefaultHeaders() {
+  List<AbfsHttpHeader> createDefaultHeaders(ApiVersion xMsVersion) {
```

Comment:
1. Can you add a Javadoc here?
2. as most invocations use the xMsVersion value, why not retain the original signature as an overloaded invocation -avoids having to change so much of the code.
3. is this package private just for testing? if so, let's mark it as @VisibleForTesting